lib/Serialization/DeserializeSIL.cpp

```diff
@@ -598,6 +624,15 @@ SILFunction *SILDeserializer::readSILFunction(DeclID FID,
   if (fn->empty() && errorIfEmptyBody)
     return nullptr;
 
+  // Check that there are no unresolved forward definitions of opened
+  // archetypes.
+  for (auto &KV : OpenedArchetypesTracker.getOpenedArchetypeDefs()) {
+    if (!KV.getSecond() || isa<GlobalAddrInst>(KV.getSecond())) {
+      llvm_unreachable(
+          "All forward definitions of opened archetypes should be resolved");
+    }
+  }
```

Comment:
Then at least move it into a helper function. In general, these sorts of predicate loops are not LLVM style:

http://llvm.org/docs/CodingStandards.html#turn-predicate-loops-into-predicate-functions